src/promptflow/promptflow/batch/_python_executor_proxy.py

```diff
@@ -32,7 +30,6 @@ def create(
     ) -> "PythonExecutorProxy":
         # TODO: Raise error if connections is None
         flow_executor = FlowExecutor.create(flow_file, connections, working_dir, storage=storage, raise_ex=False)
-        signal.signal(signal.SIGINT, signal_handler)
```

Comment:
Why put it outside? Do we have any special reasons?

Reply:
Offline synced. Added TODO.